### ISSUE TITLE
[PS-33] updating the list of URI on broacaster-side  takes an abnormally long time.

### ISSUE THREAD

**Issue by FranckUltima** [livepeer/go-livepeer#2765]
after changing server and updating the URI via livepeer_cli, after more than 3 days the new server still only receives test streams, but jobs from livepeer.inc still arrive only on the old server. 

It seems that updating the list of URIs at the broadcaster level takes an abnormally long time.

<sub>[PS-33](https://linear.app/livepeer/issue/PS-33/updating-the-list-of-uri-on-broacaster-side-takes-an-abnormally-long)</sub>

**Comment by thomshutt** [comment]
@FranckUltima could we please have the old and new URIs to investigate?

**Comment by Wendy-Utopia** [comment]
old URI : https://81.0.246.88:8935
New URI :  https://utopia-node.xyz:8953

Updated Mars 3. 

Old URI still receiving work, new one only test streams.

Thank for your Help @thomshutt

**Comment by Wendy-Utopia** [comment]
Update. 
Since 3 weeks, some livepeer.inc nodes still not updated their URI list. 
MDW is up to date (receiving job on new server)
LON and FRA still not updated. (Receiving job on old server)
I m using only one server. no geodns.

**Comment by FranckUltima** [comment]
problem seems to have been solved after the redeploy of the broadcaster's nodes this afternoon. 
But there is always an issue. 
it s  abnormal that broadcaster's nodes need to be restarted to update there orch URI list.

**Comment by thomshutt** [comment]
@FranckUltima Agreed, we're going to be prioritising this and other O issues over the next month

**Comment by 0xVires** [comment]
Any update on this issue?

I think I'm currently experiencing this with my testing O: Changed the URI (more specifically the port) 3 days ago and I'm not getting any work since then besides the test streams.

**Comment by thomshutt** [comment]
@0xVires I suspect the Broadcaster cache isn't updating properly in the case of URI changes (which is why we see a Broadcaster restart causing an update). I did a bit of digging on Friday and didn't come up with anything, but will try to figure it out this week

**Comment by yondonfu** [comment]
A few things worth keeping in mind while investigating this issue:

- If B is using on-chain discovery (the default), then the relevant code is [here](https://github.com/livepeer/go-livepeer/blob/master/discovery/db_discovery.go). The [ServiceRegistryWatcher](https://github.com/livepeer/go-livepeer/blob/master/eth/watchers/serviceRegistryWatcher.go) should be responsible for monitoring the ServiceRegistry contract for updates to an O's service URI which is then cached in B's DB.
- If B is using webhook discovery (configured using `-orchWebhookUrl`), then the relevant code is [here](https://github.com/livepeer/go-livepeer/blob/master/discovery/wh_discovery.go). The O webhook discovery implementation within Studio can be found [here](https://github.com/livepeer/studio/blob/master/packages/api/src/middleware/subgraph.ts) and it returns cached responses from the Livepeer subgraph

Worth looking into whether there are any issues with the caching logic used in any of the code paths for on-chain/webhook discovery.

**Comment by ad-astra-video** [comment]
My test broadcaster just experienced this.  Pon changed his port and my broadcaster did not look to the new port until restarted.

**Comment by FranckUltima** [comment]
I have a question: if the update on the broadcasters' side of the URI lists needs to wait for a server restart, which could result in a delay of several weeks before an orchestrator that has changed its URI starts receiving new jobs (apart from test streams), is this also the case for stakes? I mean, if an orchestrator with a stake of 2000 LPT receives a new stake of 50000 LPT, should we anticipate a potential delay of several weeks while waiting for the broadcasters to reboot their servers? Or is this independent of the URI update issue, and the new stake is immediately taken into account?
